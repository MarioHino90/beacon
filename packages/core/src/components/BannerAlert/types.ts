import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
