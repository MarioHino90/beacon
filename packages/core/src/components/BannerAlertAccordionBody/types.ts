export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
