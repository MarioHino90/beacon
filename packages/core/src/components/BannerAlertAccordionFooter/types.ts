export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
