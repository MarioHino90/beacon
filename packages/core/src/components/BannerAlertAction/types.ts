import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
