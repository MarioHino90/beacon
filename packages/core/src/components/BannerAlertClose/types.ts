import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
