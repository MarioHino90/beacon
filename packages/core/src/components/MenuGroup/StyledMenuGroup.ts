import styled, { css } from "styled-components";
import { getColor, getSpacing } from "@braze/beacon-styling";

import { StyledHeading } from "../Heading";
import { StyledMenuButton } from "../MenuButton";

import { StyledMenuGroupProps } from "./types";

export const StyledMenuGroup = styled.div<StyledMenuGroupProps>`
  width: 100%;

  &:not(:last-of-type) {
    padding-bottom: ${getSpacing("md")};
    margin-bottom: ${getSpacing("md")};
    border-bottom: 1px solid ${getColor("gray", 100)};
  }

  & > ${StyledHeading} {
    padding: ${getSpacing("md")} ${getSpacing("xl")} ${getSpacing("sm")};
    color: ${getColor("gray", 500)};
    text-transform: uppercase;
  }

  ${(props) =>
    props.hasHeading &&
    css`
      &:last-of-type {
        margin-bottom: ${getSpacing("sm")};
      }

      & > ${StyledMenuButton} {
        padding-left: ${getSpacing("xxl")};
      }
    `};
`;
