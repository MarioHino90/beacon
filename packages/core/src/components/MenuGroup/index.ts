export { MenuGroup } from "./MenuGroup";

export type { MenuGroupProps } from "./types";
