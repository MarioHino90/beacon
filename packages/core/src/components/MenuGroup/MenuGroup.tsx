import React from "react";
import classNames from "classnames";

import { Heading } from "../Heading";

import { MenuGroupProps } from "./types";
import { StyledMenuGroup } from "./StyledMenuGroup";

export const MenuGroup = ({
  children,
  className,
  heading,
  ...props
}: MenuGroupProps): React.ReactElement => (
  <StyledMenuGroup
    {...props}
    className={classNames(["bcl-menu-group", className])}
    hasHeading={!!heading}
  >
    {heading && <Heading level={6}>{heading}</Heading>}
    {children}
  </StyledMenuGroup>
);

MenuGroup.displayName = "MenuGroup";
