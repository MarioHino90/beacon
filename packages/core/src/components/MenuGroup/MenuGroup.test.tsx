import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { MenuGroupProps } from "./types";
import { MenuGroup } from "./MenuGroup";

faker.seed(1603838448107);

describe("MenuGroup", () => {
  let props: MenuGroupProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<MenuGroup {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snapshot when 'heading' is defined", () => {
    const wrapper = shallow(
      <MenuGroup {...props} heading={faker.random.words()} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
