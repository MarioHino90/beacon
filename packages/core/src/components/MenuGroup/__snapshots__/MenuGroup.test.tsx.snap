// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuGroup should match the snapshot 1`] = `
<StyledMenuGroup
  className="bcl-menu-group Gorgeous"
  hasHeading={false}
  id="tertiary"
>
  bypass
</StyledMenuGroup>
`;

exports[`MenuGroup should match the snapshot when 'heading' is defined 1`] = `
<StyledMenuGroup
  className="bcl-menu-group Fish"
  hasHeading={true}
  id="project"
>
  <Heading
    level={6}
  >
    mindshare bottom-line dot-com
  </Heading>
  Dinar
</StyledMenuGroup>
`;
